### PR TITLE
Add support for partial releases

### PR DESCRIPTION
**Pull request by thegostep**
Invariants violated by this update:
- `releaseStake()` can be called multiple times per round while still allowing to call `destroyStake()` once
- StakeReleased event will emit _releaseAmt instead of etherReward
- StakeReleased event can emit multiple times per round

### CODE REVIEW

File: contracts/NumeraireDelegateV4.sol [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+        - StakeReleased event will emit _releaseAmt instead of etherReward
+        - StakeReleased evetn can be called multiple times per round
+    */
+
+    // Numerai calls this function to release staked tokens when the staked predictions were successful
+    function releaseStake(address _staker, bytes32 _tag, uint256 _releaseAmt, uint256 _tournamentID, uint256 _roundID, bool _successful) onlyOwner stopInEmergency returns (bool ok) {
+        var round = tournaments[_tournamentID].rounds[_roundID];
+        var stake = round.stakes[_staker][_tag];
+        var originalStakeAmount = stake.amount;
+
```
Comment (kumikoda) [inline contracts/NumeraireDelegateV4.sol:40]: do we need to `require(_releaseAmount <= stake.amount)`?

Comment (thegostep) [reply contracts/NumeraireDelegateV4.sol:40]: it is implicit to `stake.amount = safeSubtract(originalStakeAmount, _releaseAmt);`

File: contracts/NumeraireDelegateV4.sol [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    // Numerai calls this function to release staked tokens when the staked predictions were successful
+    function releaseStake(address _staker, bytes32 _tag, uint256 _releaseAmount, uint256 _tournamentID, uint256 _roundID, bool _successful) onlyOwner stopInEmergency returns (bool ok) {
+        var round = tournaments[_tournamentID].rounds[_roundID];
+        var stake = round.stakes[_staker][_tag];
+        var originalStakeAmount = stake.amount;
+
+        require(stake.amount > 0);
+        require(!stake.resolved);
+        require(round.resolutionTime <= block.timestamp);
```
Comment (BryantLuu) [inline contracts/NumeraireDelegateV4.sol]: @kumikoda ^ This line scared me when reviewing the contract code. This means stakes are permanently locked if we don't resolve by 3pm right now.

File: contracts/NumeraireDelegateV4.sol [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
```
Comment (BryantLuu) [inline contracts/NumeraireDelegateV4.sol:44]: @kumikoda @thegostep 
can we add something like
```
if (_releaseAmount == 0) {
    _releaseAmount = stake.amount
}
```

We want control for releasing a partial amount, but I also don't want to need to know the exact amount to 0 out the stake and risk leaving ETH unreleased and it needs to be 0 to set resolved to true.

**Comment by kumikoda** [comment]
@BryantLuu @thegostep let's officially abandon this approach. 

I tried to spec out what the backend changes look like if we had multiple stake resolutions and it was really messy. I think the #79 is actually a much better solution

**Pull request closed without merging**